src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -31,19 +35,51 @@ public void init() {
   }
 
   @Override
-  // Get a specific user. Return null if not found. TODO: User Validation
+  // Get a specific user. Return the currently signed in user if no userID is
+  // supplied.
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGithub = request.getParameter("githubID");
-    Key userKey = KeyFactory.createKey(PersonBuilder.ENTITY_NAME, userGithub);
-    System.out.println(userKey);
 
+    String userID =
+        request.getParameter("userID") != null
+            ? request.getParameter("userID")
+            : AuthServlet.getAuthorizedUser().getUserId();
+
+    Key userKey;
+    try {
+      userKey = KeyFactory.createKey(PersonBuilder.ENTITY_NAME, userID);
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage() + "/n/n" + e.getStackTrace(),
+          HttpServletResponse.SC_BAD_REQUEST,
+          "An error occured while getting the User's profile");
+      return;
+    }
+
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(userKey);
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage() + "/n/n" + e.getStackTrace(),
+          HttpServletResponse.SC_BAD_REQUEST,
+          "The user requested could not be found on the server."
+              + "Please ensure that the user has an account with us.");
+      return;
+    }
     PersonObject userObject;
     try {
-      userObject = new PersonBuilder().buildPersonObject(datastore.get(userKey));
-      System.out.println(userObject);
+      userObject = new PersonBuilder().buildPersonObject(userEntity);
     } catch (Exception e) {
-      e.printStackTrace();
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage()
+              + "/n/n"
+              + e.getStackTrace()
+              + "/n/n PersonObject could not be instantiated from Person Entity",
+          HttpServletResponse.SC_BAD_REQUEST,
+          "User could not be instatiated in the Server");
       return;
     }
```

Comment:
See comment above about appropriate HTTP status codes. I'm thinking that this is another case where a 5xx error would be most appropriate.